flow/project.py

```diff
@@ -369,12 +512,12 @@ def cmd(self):
 
     def set_status(self, value):
         "Store the operation's status."
-        self.job._project.document.setdefault('_status', dict())[self.id] = int(value)
+        self.jobs[0]._project.document.setdefault('_status', dict())[self.id] = int(value)
```

Comment:
It is only using the `job` to get to the project. Assuming that all jobs come from the same project this is fine. The problem more comes in with changing aggregates if a project adds or removes jobs, or the aggregation is not deterministic.

Reply:
@atravitz We assume here that all the jobs in the `JobOperation` instance are from the same project hence we just fetch the project from a job.

Reply:
@b-butler, For instance, we submit the following operation
```
@aggregate.groupsof(2)
@FlowProject.operation
def foo(*jobs):
    import time
    time.sleep(5)
    print('slept well')
```

And in the meanwhile we delete a job which is present in the aggregated which is not yet submitted.
This will throw the following error which says the following for that particular aggregate.

> Using environment configuration: DefaultSlurmEnvironment
> ERROR: Encountered error during program execution: 'Did not find job with id '670d5659b2965b80171e24047ff66a40'.'
> 
> Execute with '--show-traceback' or '--debug' to get more information.
>